.devops/nix/package.nix

```diff
@@ -216,7 +235,10 @@ effectiveStdenv.mkDerivation (
         # Should likely use `rocmPackages.clr.gpuTargets`.
         "-DAMDGPU_TARGETS=gfx803;gfx900;gfx906:xnack-;gfx908:xnack-;gfx90a:xnack+;gfx90a:xnack-;gfx940;gfx941;gfx942;gfx1010;gfx1012;gfx1030;gfx1100;gfx1101;gfx1102"
       ]
-      ++ optionals useMetalKit [ (lib.cmakeFeature "CMAKE_C_FLAGS" "-D__ARM_FEATURE_DOTPROD=1") ]
+      ++ optionals useMetalKit [
+        (lib.cmakeFeature "CMAKE_C_FLAGS" "-D__ARM_FEATURE_DOTPROD=1")
```

Comment:
I was going to mention this before and now that there's a merge conflict I have a reason to: multiple `-DCMAKE_C_FLAGS` options aren't going to compose. I think we should move those to `NIX_CFLAGS_COMPILE`

Reply:
Are there any other uses of `CMAKE_C_FLAGS` currently? Also, could you elaborate on moving this option into `NIX_CFLAGS_COMPILE`? 

Right now, thinking of something like

```
let
  cmakeCFlags = lib.optionalString useMetalKit [ "-D__ARM_FEATURE_DOTPROD=1" ];
# ...
in {
# ...

cmakeFlags = # ...rest of section
  ++ lib.cmakeFeature "CMAKE_C_FLAGS" cmakeCFlags;
# ...
}
```

Which would allow other backends to configure `CMAKE_C_FLAGS` via appending to `cmakeCFlags`

Reply:
`NIX_CFLAGS_COMPILE` is an environment variable interpreted by nixpkgs' `cc-wrapper`. You can just pass it to `mkDerivation`: `NIX_CFLAGS_COMPILE = [ "-DA=1" "-DB=2" "-D__ARM_FEATURE_DOTPROD=1" ]`

> Are there any other uses of CMAKE_C_FLAGS currently?

I thought that was the merge conflict?

Reply:
I guess I was confused, it must've been just a change in formatting

Reply:
ah, good to know about that variable. Merge conflict was from the following line (blas support) being deleted in master, likely a merge conflict line since I split the preceding line onto two lines